corehq/form_processor/backends/sql/update_strategy.py

```diff
@@ -220,7 +220,7 @@ def _apply_attachments_action(self, attachment_action, xform):
             if att.is_delete:
                 if name in current_attachments:
                     self.case.track_delete(current_attachments[name])
-            else:
+            elif att.attachment_src:
```

Comment:
Why would there be an attachment in `attachment_action.attachments` if it is not being deleted or have a valid reference to a form attachment? This code will ignore those items, but that seems like a bug. Maybe there should be a soft assert in an `else` clause?